From: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Date: Tue, 4 Feb 2025 12:30:32 +0100
Subject: [PATCH] Reapply "serial: 8250: Revert "drop lockdep annotation from
 serial8250_clear_IER()""

This reverts commit 244eb5c6ec62ccab59ecac1f4815bb33130c423a.

Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
---
 drivers/tty/serial/8250/8250_port.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/tty/serial/8250/8250_port.c
+++ b/drivers/tty/serial/8250/8250_port.c
@@ -724,6 +724,9 @@ static void __serial8250_clear_IER(struc
 
 static inline void serial8250_clear_IER(struct uart_8250_port *up)
 {
+	/* Port locked to synchronize UART_IER access against the console */
+	lockdep_assert_held_once(&up->port.lock);
+
 	__serial8250_clear_IER(up);
 }
 
